src/main/java/com/microsoft/graph/content/BatchResponseContent.java

```diff
@@ -0,0 +1,227 @@
+        this.batchResponseProtocol = batchResponse.protocol();
+        this.batchResponseRequest = batchResponse.request();
+        this.apiErrorMappings = apiErrorMappings == null ? new HashMap<>() : new HashMap<>(apiErrorMappings);
+    }
+    /**
+     * Gets the responses of the batch request.
+     * @return The responses of the batch request.
+     */
+    @Nonnull
+    public CompletableFuture<HashMap<String, Response>> getResponses() {
+        HashMap<String, Response> responses = new HashMap<>();
+        jsonBatchResponseObject = jsonBatchResponseObject != null ? jsonBatchResponseObject : getBatchResponseContent().join();
+        if (jsonBatchResponseObject != null) {
+            JsonElement responsesElement = jsonBatchResponseObject.get(CoreConstants.BatchRequest.RESPONSES);
+            if (responsesElement != null && responsesElement.isJsonArray()) { //ensure "responses" is not null and is an array.
+                JsonArray responsesArray = responsesElement.getAsJsonArray();
+                for (JsonElement responseElement : responsesArray) {
+                    responses.put(responseElement.getAsJsonObject().get("id").getAsString(), getResponseFromJsonObject(responseElement));
+                }
+            }
+        }
+        return CompletableFuture.completedFuture(responses);
+    }
+    /**
+     * Gets the status codes of the responses of the batch request.
+     * @return The status codes of the responses of the batch request.
+     */
+    @Nonnull
+    public CompletableFuture<HashMap<String, Integer>> getResponsesStatusCode() {
+        HashMap<String, Integer> statusCodes = new HashMap<>();
+        jsonBatchResponseObject = jsonBatchResponseObject != null ? jsonBatchResponseObject : getBatchResponseContent().join();
+        if (jsonBatchResponseObject != null) {
+            JsonElement responsesElement = jsonBatchResponseObject.get(CoreConstants.BatchRequest.RESPONSES);
+            if (responsesElement != null && responsesElement.isJsonArray()) { //ensure "responses" is not null and is an array.
+                JsonArray responsesArray = responsesElement.getAsJsonArray();
+                for (JsonElement responseElement : responsesArray) {
+                    statusCodes.put(responseElement.getAsJsonObject().get("id").getAsString(), getStatusCodeFromJsonObject(responseElement));
+                }
+            }
+        }
+        return CompletableFuture.completedFuture(statusCodes);
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @return The response within the batch response via specified id, null if not found.
+     */
+    @Nullable
+    public CompletableFuture<Response> getResponseById(@Nonnull String requestId) {
+        jsonBatchResponseObject = jsonBatchResponseObject != null ? jsonBatchResponseObject : getBatchResponseContent().join();
```

Comment:
I think this should be an empty check rather than a null check, the annotation should be plenty enough to ensure it's not null, at least that's the expectation. I think an empty check would be better here.

Reply:
I guess a more general question, should we always require null checks even when a param is annotated as nonnull? I usually get an IDE suggestion to delete the requireNonNull statement since the annotation should ensure nonnull.

Reply:
Interesting, I get the opposite suggestion. My understanding was that the nonnull annotation doesn't do anything at runtime. Hence the requirement to "manually" enforce the defensive programming. Which is a bit sad from an Aspect Oriented Programming perspective.

Now, after doing some research, it seems that [intelliJ does some dark magic](https://stackoverflow.com/questions/40847472/why-nonnull-annotation-checked-at-runtime) which is a bit problematic because the byte code you're testing locally is not going to be the same we're sending to the customers...

My suggestion: disable the option they are referring to, so you can test code that's identical to what we're shipping and use Objects.requireNonNull (+ string empty/blank when necessary)

If I missed anything, please let me know.